enterprise/api/v1/views/enterprise_customer.py

```diff
@@ -487,3 +487,32 @@ def unlink_users(self, request, pk=None):  # pylint: disable=unused-argument
                     raise UnlinkUserFromEnterpriseError(msg) from exc
 
         return Response(status=HTTP_200_OK)
+
+    @action(methods=['post'], detail=True, permission_classes=[permissions.IsAuthenticated, IsUseremailInRequest])
+    def unlink_self(self, request, pk=None):  # pylint: disable=unused-argument
+        """
+        Unlink request user from the enterprise.
+        """
+        serializer = serializers.EnterpriseCustomerUnlinkSelfUserSerializer(
+            data=request.data
+        )
+
+        serializer.is_valid(raise_exception=True)
+        enterprise_customer = self.get_object()
+        user_email = serializer.data.get('user_email')
+        is_relinkable = serializer.data.get('is_relinkable', True)
+
+        try:
+            models.EnterpriseCustomerUser.objects.unlink_user(
+                enterprise_customer=enterprise_customer, user_email=user_email, is_relinkable=is_relinkable
+            )
```

Comment:
We need to have this view only be able to unlink the user making the request.  So I don't think we need `user_email` in the request payload. Instead, we can use `request.user.email` as the value to pass into `unlink_user()`.

Reply:
I see you're basically enforcing that in the `IsUseremailInRequest` permission class. I think you can get rid of that permission class and just go with 
```
models.EnterpriseCustomerUser.objects.unlink_user(
    enterprise_customer=enterprise_customer, user_email=request.user.email, is_relinkable=is_relinkable
)
```
here in the view.